test_conformance/math_brute_force/binary_half.cpp

```diff
@@ -0,0 +1,791 @@
+                            correct2 = ref_func(s[j], 0.0);
+                            correct3 = ref_func(s[j], -0.0);
+                        }
+
+                        if (skipNanInf)
+                        {
+                            // Note: no double rounding here.  Reference
+                            // functions calculate in single precision.
+                            if (IsFloatInfinity(correct) || IsFloatNaN(correct)
+                                || IsFloatInfinity(correct2)
+                                || IsFloatNaN(correct2))
+                                continue;
+                        }
+
+                        auto check_error = [&]() {
+                            err2 = Ulp_Error_Half(q[j], correct2);
+                            err3 = Ulp_Error_Half(q[j], correct3);
+                            fail = fail
+                                && ((!(fabsf(err2) <= ulps))
+                                    && (!(fabsf(err3) <= ulps)));
+                            if (fabsf(err2) < fabsf(err)) err = err2;
+                            if (fabsf(err3) < fabsf(err)) err = err3;
+                        };
+                        check_error();
+
+                        // retry per section 6.5.3.4
+                        if (IsHalfSubnormal(
+                                cl_half_from_float(correct2, CL_HALF_RTE))
+                            || IsHalfSubnormal(
+                                cl_half_from_float(correct3, CL_HALF_RTE)))
+                        {
+                            if (fail && isNextafter)
+                            {
+                                correct2 =
+                                    reference_nextafterh(s[j], 0.0, false);
+                                correct3 =
+                                    reference_nextafterh(s[j], -0.0, false);
+                                check_error();
+                            }
+
+                            fail = fail && (test != 0.0f);
+                            if (!fail) err = 0.0f;
+                        }
+
+                        // allow to omit denorm values for platforms with no
+                        // denorm support for nextafter
+                        if (fail && (isNextafter)
+                            && (correct <= cl_half_to_float(0x3FF))
+                            && (correct >= cl_half_to_float(0x83FF)))
+                        {
+                            fail = fail && (q[j] != p2[j]);
+                            if (!fail) err = 0.0f;
+                        }
+                    }
+                }
+
+                if (fabsf(err) > tinfo->maxError)
+                {
+                    tinfo->maxError = fabsf(err);
+                    tinfo->maxErrorValue = s[j];
+                    tinfo->maxErrorValue2 = s2[j];
+                }
+                if (fail)
+                {
+                    vlog_error("\nERROR: %s%s: %f ulp error at {%a (0x%0.4x), "
```

Comment:
This introduces more Wformat errors:
```
error: '0' flag ignored with precision and ‘%x’ gnu_printf format
```

From the `printf` man page:
> If a precision is given with a numeric conversion (d, i, o, u, x, and X), the 0 flag is ignored.

This affects other `_half.cpp` files too.

Reply:
Done